content/en/build/protocol/building-on-protocol-layer.md

```diff
@@ -0,0 +1,191 @@
+weight: 10
+---
+
+If you are building tooling such as wallets, exchanges, and developer tools, you may wish to interact directly with Filecoin’s protocol layer. This section includes resource pages for building directly on the protocol layer, and quickstart guides for [curl](#curl-quickstart) and [NodeJS].
+
+See [Available Networks](https://docs.filecoin.io/networks/overview/) for information on test networks and Mainnet. 
+
+## Curl Quickstart
+
+This quickstart aims to give you a taste of working directly with the Filecoin protocol in less than 5 minutes. We will use JSON RPC API calls along with [curl](https://curl.se/), a command-line utility that’s pre-installed into most Linux distributions, to read public chain data from Filecoin.
+
+In this tutorial, you will learn how to: 
+* [Get current block height](#get-current-block-height)
+* [Get tipset by height and find block CID](#get-tipset-by-height-and-find-block-cid)
+* [Get block details](#get-block-details)
+
+**A note on authentication:** Publicly hosted nodes only support unsigned methods that don’t require an API token. To use all of the methods, you’ll need to host your own node or get access to a [dedicated hosted node](https://lotus.filecoin.io/lotus/developers/glif-nodes/). For more details, see [Lotus API access](https://lotus.filecoin.io/reference/basics/api-access/).
+
+### Get current block height
+
+Let’s start by getting the current block height of the chain. This can be found by reading the current highest tipset using the `Filecoin.ChainHead` method:
+
+```
+curl -X POST 'https://api.node.glif.io' -H 'Content-Type: application/json' --data '{"jsonrpc":"2.0","id":1,"method":"Filecoin.ChainHead","params":[]}' -H "Accept: application/json" -s | jq
+```
+
+
+This call will return plenty of information, but we are only interested in the `Height` parameter:
+
+
+```
+{
+    "result": {
+        "Cids": [
+            {
+                "/": "bafy2bzacedr7iporlwiwnpx57mnanrncfa7aegyuoduosyk36mcesqvpoyrns"
+            },
+            ...
+        ],
+        "Height": 1956488,
+        "Blocks": [
+            { ... }
+        ]
+    },
+    "id": 1,
+    "jsonrpc": "2.0"
+}
+```
+
+At the time of writing, the block height is 1956488. You can confirm the current block height via any block explorer, such as [Filfox](https://filfox.info/en).
+
+## Get tipset by height and find block CID
+
+Now we have the height, we can query the node to find the tipset by this height. A tipset is a set of blocks that each have the same height and parent tipset. Filecoin’s consensus mechanism allows zero or multiple blocks at each epoch; blocks from the same epoch are assembled into tipsets.
+
+Use `ChainGetTipSetByHeight` to get the tipset at a given block height:
+
+```
+curl -X POST 'https://api.node.glif.io' -H 'Content-Type: application/json' --data '{"jsonrpc":"2.0","id":1,"method":"Filecoin.ChainGetTipSetByHeight","params":[1918961]}' -H "Accept: application/json" -s | jq
```

Comment:
I test this command, it does not work as expected and throw an error `wrong param count (method 'Filecoin.ChainGetTipSetByHeight'): 1 != 2`。
I think it is related to the Glif node only maintaining the latest 2000 blocks. Maybe we need add a note here.